Yoast/wordpress-seo
# Fix react-intl dev warnings

## Summary

This PR can be summarized in the following changelog entry:

* Fixes react-intl warnings in development.

## Relevant technical choices:

* Implemented asynchronous loading of react-intl's locale data via webpack.
* Implemented `babel-eslint` to support the experimental import syntax.

## Test instructions

This PR can be tested by following these steps:

* Before checking out this PR check for the react-intl errors on the following pages:
  1. Dashboard Widget on the WordPress dashboard
  2. Help center on Yoast SEO settings pages and post edit page (classic editor)
  3. Content analysis on post/page/taxonomy edit pages
* Check out this PR and build `wordpress-seo`
* Check the components from step 1 again, make sure no errors occur.
* Select a different language that is fully translated (i.e. Dutch)
* Check the components from step 1 again, make sure no errors occur.

## Quality ensurance


Fixes #8518
